src/builtins.cpp

```diff
@@ -1265,3 +1266,32 @@ void ispc::DefineStdlib(SymbolTable *symbolTable, llvm::LLVMContext *ctx, llvm::
         yy_delete_buffer(strbuf);
     }
 }
+
+#ifdef ISPC_XE_ENABLED
+std::string ispc::mangleSPIRVBuiltin(const llvm::Function &func) {
+    Assert(func.getName().startswith("__spirv_") && "wrong argument: spirv builtin is expected");
+    const std::string fnName = func.getName().str();
+    std::string mangledName;
+
+    if (func.getName().startswith("__spirv_SpecConstant")) {
+        std::vector<llvm::Type *> tyArgs;
+        for (const auto &arg : func.args()) {
+            tyArgs.push_back(arg.getType());
+        }
+        mangleOpenClBuiltin(fnName, tyArgs,
+#if ISPC_LLVM_VERSION == ISPC_LLVM_15_0
+                            // spirv builtins doesn't have pointer arguments
+                            {},
+#endif
+                            mangledName);
+    } else {
+        mangleOpenClBuiltin(fnName, func.getArg(0)->getType(),
```

Comment:
Please figure out when this code triggers with more than one argument and why we pass only the first one. This code looks suspicious.

Reply:
Failing lit tests with `Assert(func.arg_size() > 1 && "Too many arguments to SPIRV builtin");`:
```text
Failed Tests (19):
  ispc :: assume_uniform_xe.ispc
  ispc :: load_store_vectorizer_loopunroll.ispc
  ispc :: print_errors_xe.ispc
  ispc :: xe_assert.ispc
  ispc :: xe_barrier.ispc
  ispc :: xe_diarglist.ispc
  ispc :: xe_externC.ispc
  ispc :: xe_gather_coalescing_loop.ispc
  ispc :: xe_half_support.ispc
  ispc :: xe_kernels_spec.ispc
  ispc :: xe_kernels_spec_addrspace.ispc
  ispc :: xe_prefetchw.ispc
  ispc :: xe_regcall.ispc
  ispc :: xe_short_circuit.ispc
  ispc :: xe_spirv_ocl_mangle.ispc
  ispc :: xe_task_count.ispc
  ispc :: xe_yarpgen_3330730858.ispc
  ispc :: xe_yarpgen_4096674170.ispc
  ispc :: xehpc_vec_add.ispc
```

Reply:
All are failing at mangling function `__spirv_BuiltInWorkgroupId`.

Reply:
The only uses only have one argument, however:
```cpp
builtins/target-xe.ll:declare i64 @__spirv_BuiltInWorkgroupId(i32 %dim)
builtins/target-xe.ll:  %gr_id_x_64 = call i64 @__spirv_BuiltInWorkgroupId(i32 0)
builtins/target-xe.ll:  %gr_id_y_64 = call i64 @__spirv_BuiltInWorkgroupId(i32 1)
builtins/target-xe.ll:  %gr_id_z_64 = call i64 @__spirv_BuiltInWorkgroupId(i32 2)
builtins/target-xe.ll:  %gr_id_64 = call i64 @__spirv_BuiltInWorkgroupId(i32 $1)
```

Reply:
I'm confused, what are the other arguments?

Reply:
Corrected assertion: `Assert(func.arg_size() <= 1 && "Too many arguments to SPIRV builtin");`

Reply:
No failures for lit tests now, so it looks like it would be safe to merge the cases.